src/wooting-usb.c

```diff
@@ -63,108 +70,116 @@ static uint16_t getCrc16ccitt(const uint8_t *buffer, uint16_t size) {
+  wooting_usb_meta->v2_interface = true;
 }
 
 static void set_meta_wooting_two_le(void) {
-  wooting_usb_meta.model = "Wooting Two Lekker Edition";
-  wooting_usb_meta.device_type = DEVICE_KEYBOARD;
-  wooting_usb_meta.max_rows = WOOTING_RGB_ROWS;
-  wooting_usb_meta.max_columns = WOOTING_TWO_RGB_COLS;
-  wooting_usb_meta.led_index_max = WOOTING_TWO_KEY_CODE_LIMIT;
-  wooting_usb_meta.v2_interface = true;
+  wooting_usb_meta->model = "Wooting Two Lekker Edition";
+  wooting_usb_meta->device_type = DEVICE_KEYBOARD;
+  wooting_usb_meta->max_rows = WOOTING_RGB_ROWS;
+  wooting_usb_meta->max_columns = WOOTING_TWO_RGB_COLS;
+  wooting_usb_meta->led_index_max = WOOTING_TWO_KEY_CODE_LIMIT;
+  wooting_usb_meta->v2_interface = true;
 }
 
 static void set_meta_wooting_two_he(void) {
-  wooting_usb_meta.model = "Wooting Two HE";
-  wooting_usb_meta.device_type = DEVICE_KEYBOARD;
-  wooting_usb_meta.max_rows = WOOTING_RGB_ROWS;
-  wooting_usb_meta.max_columns = WOOTING_TWO_RGB_COLS;
-  wooting_usb_meta.led_index_max = WOOTING_TWO_KEY_CODE_LIMIT;
-  wooting_usb_meta.v2_interface = true;
+  wooting_usb_meta->model = "Wooting Two HE";
+  wooting_usb_meta->device_type = DEVICE_KEYBOARD;
+  wooting_usb_meta->max_rows = WOOTING_RGB_ROWS;
+  wooting_usb_meta->max_columns = WOOTING_TWO_RGB_COLS;
+  wooting_usb_meta->led_index_max = WOOTING_TWO_KEY_CODE_LIMIT;
+  wooting_usb_meta->v2_interface = true;
 }
 
 static void set_meta_wooting_60he(void) {
-  wooting_usb_meta.model = "Wooting 60HE";
-  wooting_usb_meta.device_type = DEVICE_KEYBOARD_60;
-  wooting_usb_meta.max_rows = WOOTING_RGB_ROWS;
-  wooting_usb_meta.max_columns = 14;
-  wooting_usb_meta.led_index_max = WOOTING_TWO_KEY_CODE_LIMIT;
-  wooting_usb_meta.v2_interface = true;
+  wooting_usb_meta->model = "Wooting 60HE";
+  wooting_usb_meta->device_type = DEVICE_KEYBOARD_60;
+  wooting_usb_meta->max_rows = WOOTING_RGB_ROWS;
+  wooting_usb_meta->max_columns = 14;
+  wooting_usb_meta->led_index_max = WOOTING_TWO_KEY_CODE_LIMIT;
+  wooting_usb_meta->v2_interface = true;
 }
 
 static void set_meta_wooting_60he_arm(void) {
-  wooting_usb_meta.model = "Wooting 60HE (ARM)";
-  wooting_usb_meta.device_type = DEVICE_KEYBOARD_60;
-  wooting_usb_meta.max_rows = WOOTING_RGB_ROWS;
-  wooting_usb_meta.max_columns = 14;
-  wooting_usb_meta.led_index_max = WOOTING_TWO_KEY_CODE_LIMIT;
-  wooting_usb_meta.v2_interface = true;
+  wooting_usb_meta->model = "Wooting 60HE (ARM)";
+  wooting_usb_meta->device_type = DEVICE_KEYBOARD_60;
+  wooting_usb_meta->max_rows = WOOTING_RGB_ROWS;
+  wooting_usb_meta->max_columns = 14;
+  wooting_usb_meta->led_index_max = WOOTING_TWO_KEY_CODE_LIMIT;
+  wooting_usb_meta->v2_interface = true;
 }
 
 WOOTING_USB_META *wooting_usb_get_meta() {
   // We want to initialise the struct to the default values if it hasn't been
   // set
-  if (wooting_usb_meta.model == NULL) {
+  if (wooting_usb_meta->model == NULL) {
     reset_meta();
   }
 
-  return &wooting_usb_meta;
+  return wooting_usb_meta;
 }
```

Comment:
I imagine it would be nice to have a function that can retrieve the meta of all devices, so you can get all the detail without having to iterate through and select every device and then get the meta for each

Reply:
yes but im not sure how to keep backwards compatibility with all existing software that uses the rgb sdk already. I basically used the select approach instead of the application managing which device to talk to so that old software still works as it did with a new sdk lib

Reply:
unless you mean to just hand out the pointer array of all the meta the lib keeps internally

Reply:
Going for a function which accepts the index and returns the meta function of that index